src/openms_gui/source/VISUAL/DIALOGS/TheoreticalSpectrumGenerationDialog.cpp

```diff
@@ -95,16 +113,16 @@ namespace OpenMS
     // select peptide sequence by default and update interface
     ui_->seq_type->setCurrentText("Peptide");
     seqTypeSwitch_();
-
-    // select b- and y-ions as residue types by default
-    for (size_t i = 0; i < int(CheckBox::NUMBER_OF_CHECK_BOXES); ++i)
+    for (size_t i = 0; i < check_boxes_.size(); ++i)
     {
-      if (i == int(CheckBox::B_Ions) || i == int(CheckBox::Y_Ions))
+      if (check_boxes_.at(i).state.at(0) == State::PRECHECKED) // 'state.at(0)' because seq type was just set to "Peptide"
       {
         ui_->ion_types->item(i)->setCheckState(Qt::Checked);
-        continue;
       }
-      ui_->ion_types->item(i)->setCheckState(Qt::Unchecked);
+      else
+      {
+        ui_->ion_types->item(i)->setCheckState(Qt::Unchecked);
+      }
     }
```

Comment:
should this be part of updating the UI when switching between types?

Reply:
Probably not.. Since all other settings are preserved when switching between sequence types, changing the checked ions feels inconsistent.